crates/fluvio-cluster/src/runtime/local/k8s.rs

```diff
@@ -0,0 +1,211 @@
+        K8Config::KubeConfig(kube_context) => {
+            let mut config = kube_context.config;
+            let user = User {
+                name: DEFAULT_K8S_USER_NAME.to_string(),
+                user: UserDetail {
+                    client_certificate: Some(certs.client_cert.to_string_lossy().to_string()),
+                    client_key: Some(certs.client_private_key.to_string_lossy().to_string()),
+                    ..Default::default()
+                },
+            };
+
+            let cluster = Cluster {
+                name: DEFAULT_K8S_CLUSTER_NAME.to_string(),
+                cluster: ClusterDetail {
+                    certificate_authority: Some(certs.root_cert.to_string_lossy().to_string()),
+                    server: endpoint.to_string(),
+                    ..Default::default()
+                },
+            };
+
+            let namespace = DEFAULT_K8S_NAMESPACE;
+            let context = Context {
+                name: DEFAULT_K8S_CONTEXT_NAME.to_string(),
+                context: ContextDetail {
+                    cluster: cluster.name.clone(),
+                    user: user.name.clone(),
+                    namespace: Some(namespace.to_string()),
+                },
+            };
+
+            config.current_context = context.name.clone();
+            config.put_user(user);
+            config.put_cluster(cluster);
+            config.put_context(context);
+
+            config.save()?;
+
+            info!(
+                "k8s context succesfully added  to {}",
+                config.path.to_string_lossy()
+            );
+
+            Ok(())
+        }
+    }
+}
+
+fn generate_certificates(certs_path: &Path) -> anyhow::Result<Certs> {
+    let _ = std::fs::create_dir_all(certs_path);
+    let kube_private_key = certs_path.join("kubernetes-key.pem");
+    let kube_cert = certs_path.join("kubernetes.pem");
+    let service_account_private_key = certs_path.join("service-account-key.pem");
+    let service_account_cert = certs_path.join("service-account.pem");
+    let client_private_key = certs_path.join("client-key.pem");
+    let client_cert = certs_path.join("client.pem");
+
+    let root_cert = certs_path.join("ca.pem");
+    let root_private_key = certs_path.join("ca-key.pem");
+    if !root_cert.exists() {
+        info!("regenerating k8s certificates");
+        let root = generate_root_cert(&root_private_key, &root_cert)?;
+
+        generate_cert(&root, &kube_private_key, &kube_cert)?;
+        generate_cert(&root, &service_account_private_key, &service_account_cert)?;
+        generate_cert(&root, &client_private_key, &client_cert)?;
+    }
+
+    Ok(Certs {
+        kube_private_key,
+        kube_cert,
+        service_account_private_key,
+        service_account_cert,
+        root_cert,
+        client_private_key,
+        client_cert,
+    })
+}
+
+fn generate_root_cert(private_key: &Path, public_key: &Path) -> anyhow::Result<Certificate> {
```

Comment:
Suggest change `private_key` to `private_key_path` and same for public_key.  Got confused reading code so it was key value itself

Reply:
The '_path' suffix would be needed if the type was 'String', but here type is 'Path' which excludes any confusion.